posix: Add support for arm-clang libc

galak opened the pull request:
Add CONFIG_ARMCLANG_STD_LIBC to places they are needed for things to build correct when compiling with arm-clang.

galak committed:
posix: Add support for arm-clang libc

Add CONFIG_ARMCLANG_STD_LIBC to places they are needed for things
to build correct when compiling with arm-clang.

Signed-off-by: Kumar Gala <kumar.gala@intel.com>

keith-packard commented on include/zephyr/posix/posix_types.h:
```diff
@@ -47,7 +47,7 @@ struct pthread_attr {
 	uint32_t initialized;
 };
-#if defined(CONFIG_MINIMAL_LIBC) || defined(CONFIG_PICOLIBC)
+#if defined(CONFIG_MINIMAL_LIBC) || defined(CONFIG_PICOLIBC) || defined(CONFIG_ARMCLANG_STD_LIBC)
```
Should we flip this logic so that it excludes newlib rather than including every other C library? Can @cfriedt explain why the structs are always defined but the typedef is only included for known C libraries? It seems like the struct shouldn't be used in the newlib case either...

cfriedt replied:
Newlib already includes a typedef for this, and the type that newlib defines is larger than the structure that zephyr defines. Given that the plan for POSIX is to blow some of this away and replace with more mature POSIX headers, I'm fine with this change for now.

keith-packard replied:
Yeah, I was just asking if this should be `#ifndef CONFIG_NEWLIB_LIBC` instead of listing all other C libraries. As you say, it's transient, so it probably doesn't matter.

cfriedt replied:
@galak - Keith has a good point there. Could you add another commit that modifies `tests/posix/common/testcase.yaml`?

keith-packard reviewed (commented):
This all looks like a simple extension to another C library. Did you verify the values from fcntl.h to see if they match? Picolibc needed another conditional there.

galak committed:
tests: posix: common: Add testcase for arm-clang

Add a testcase variant to cover building with arm-clang and
CONFIG_ARMCLANG_STD_LIBC set.

Signed-off-by: Kumar Gala <kumar.gala@intel.com>

galak commented:
@cfriedt, @keith-packard  can you guys take another look/approve.  thanks.

keith-packard reviewed (approved):
Yup, seems like it will work.
